### PR TITLE
Add missing Data attribute for Treemap component

### PR DESCRIPTION
**Pull request by ghost**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [ ] The package does not provide its own types, and you can not add them.
- [ ] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [ ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [ ] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, and `strictNullChecks` set to `true`.

If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

If removing a declaration:
- [ ] If a package was never on DefinitelyTyped, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

**Comment by mthmulders** [comment]
👍

**Comment by rapmue** [comment]
👍

File: types/recharts/index.d.ts [hunk 1]
```diff
@@ -700,8 +700,8 @@ export class Tooltip extends React.Component<TooltipProps> { }
 
 export interface TreemapProps {
 	data: any[];
-	width: number;
-	height: number;
+	width?: number;
+	height?: number;
```
Comment (rapmue) [inline types/recharts/index.d.ts]: @li-xinyang  Is this somewhere documented?

**Review by rapmue — COMMENTED** [review]
Do you have a documentation about this features?

File: types/recharts/index.d.ts (hunk 1 above)
Comment (li-xinyang) [reply types/recharts/index.d.ts]: https://github.com/recharts/recharts/blob/master/src/chart/Treemap.js#L193

Basically there is NOT required props for all charts component which is kind pain to use TS with all these props as **required**

**Comment by uniqueiniquity** [comment]
@mthmulders @rapmue I'll wait for reapproval on the second commit before merging.

**Comment by mthmulders** [comment]
The second commit makes sense. Indeed the props aren't required in Recharts itself.

**Review by rapmue — APPROVED** [review]

**Pull request merged**